### DISCUSSION TITLE
How to create a pore distribution manually

### DISCUSSION THREAD

**Discussion by livashu** [PMEAL/OpenPNM#2112] [Q&A] [answered]
I would like to create a pore distribution manually from experimental data without using Weibull distribution.
Is there a good way to do this?
Is there any way to read a graph created in Excel, for example?

**Comment by jgostick** [comment, marked as answer]
So you're saying that you have some like a histogram, and you want to add pores in proportion to height of each bar in your histogram.  We do not have a function for this explicitly, but it should be do-able:

``` python

import numpy as np
import matplotlib.pyplot as plt
N = 1000  # N is the total number of pores diameter you want to generate
bin_centers = [0.5, 2.1, 3.3, 4.0, 5.0]  # In what ever units you like
bin_height = [0.05, 0.2, 0.4, 0.3, 0.15]  # Normalized to sum to 1.0
r = np.zeros(N)  
i = 0
while i < N:
    bin = np.random.randint(len(bin_centers))
    if np.random.rand() < bin_height[bin]:
        r[i] = bin_centers[bin]
        i += 1
plt.hist(r)

```

**Reply by livashu** [reply]
I managed to succeed. Thank you!

**Reply by mourendong** [reply]
I also wanted to create a pore distribution from experimental data. 
As the way mentioned above, I just created the histogram, how to apply it to the size distribution of the pore network?
Can you tell me more detail about it?

**Reply by jgostick** [reply]
Once you get the values in ``r``, you assign them to your ``geometry`` object, like ``geo['pore.diameter'] = r``
